modules/oci-ocne-autoscaling/variables.tf

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2023 Oracle Corporation and/or affiliates.  All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+// Common OCI resource variables
+variable "availability_domain_id" {
+  type        = string
+  description = "The ID of the availability domain inside the `region` in which to create this deployment"
+}
+
+variable "compartment_id" {
+  type        = string
+  description = "The OCID of the compartment in which to create this deployment"
+}
+
+variable "prefix" {
+  type        = string
+  description = "A unique prefix to attach to the name of all cloud resources that are created as a part of this deployment"
+}
+
+variable "instance_shape" {
```

Comment:
This should be a map to accomodate flex. see https://github.com/oracle-terraform-modules/terraform-oci-bastion/blob/main/variables.tf#L77